### PR TITLE
*: further adaptation to PD HTTP client

### PR DESCRIPTION
**Pull request by JmPotato**
### What problem does this PR solve?

Issue Number: ref #35319, https://github.com/tikv/pd/issues/7300. Should merge after https://github.com/tikv/pd/pull/7371.

Problem Summary:

### What is changed and how it works?

- Remove `pkg/util/pdapi` package.
- Further integrate with the PD HTTP client.

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by tangenta — APPROVED** [review]

File: br/pkg/restore/split/client.go [hunk 1]
```diff
@@ -628,7 +628,7 @@ func (c *pdClient) SetStoresLabel(
 	return nil
 }
 
-func (c *pdClient) getPDAPIAddr() string {
+func (c *pdClient) getpdhttpAddr() string {
```
Comment (YuJuncen) [inline br/pkg/restore/split/client.go:631]: name it `getPDHTTPAddr`?

**Review by YuJuncen — COMMENTED** [review]
rest lgtm

File: br/pkg/restore/split/client.go (hunk 1 above)
Comment (JmPotato) [reply br/pkg/restore/split/client.go]: I think I replaced it globally mistakenly. Changed back. PTAL.

**Review by lichunzhu — APPROVED** [review]

**Review by YuJuncen — APPROVED** [review]

**Pull request merged**